crypto/src/main/java/net/corda/v5/crypto/SecureHash.java

```diff
@@ -1,109 +1,55 @@
-public final class SecureHash extends OpaqueBytes {
+public interface SecureHash {
     /**
-     * The hash algorithm.
-     */
-    private final String algorithm;
-
-   /**
-     * Construct a secure hash.
-     *
-     * @param algorithm Hashing algorithm which was used to generate the hash.
-     * @param bytes     Hash value.
+     * Hashing algorithm which was used to generate the hash.
      */
-    public SecureHash(@NotNull String algorithm, @NotNull byte[] bytes) {
-        super(bytes);
-        this.algorithm = algorithm;
-    }
-
-    static final char DELIMITER = ':';
+    @NotNull
+    String getAlgorithm();
 
+    // TODO we could get away without this, but adding it for consistency with `PrivacySalt`
     /**
-     * Creates a {@link SecureHash}.
-     * <p>
-     * This function does not validate the length of the created digest. Prefer using
-     * <code>HashingService#parse</code> for a safer mechanism for creating {@link SecureHash}es.
+     * The result bytes of the hashing operation with the specified digest algorithm. The specified digest algorithm
+     * can be acquired through {@link SecureHash#getAlgorithm()}
      */
     @NotNull
-    public static SecureHash parse(@NotNull String str) {
-        int idx = str.indexOf(DELIMITER);
-        if (idx == -1) {
-            throw new IllegalArgumentException("Provided string: $str should be of format algorithm:hexadecimal");
-        } else {
-            String algorithm = str.substring(0, idx);
-            String value = str.substring(idx + 1);
-            byte[] data = ByteArrays.parseAsHex(value);
-            return new SecureHash(algorithm, data);
-        }
-    }
+    byte[] getBytes();
 
     /**
      * Returns hexadecimal representation of the hash value.
      */
     @NotNull
-    public String toHexString() {
-        return ByteArrays.toHexString(this.getBytes());
-    }
+    String toHexString();
 
     /**
      * Returns the first specified number of hexadecimal digits of the {@link SecureHash} value.
      *
      * @param prefixLen The number of characters in the prefix.
      */
     @NotNull
-    public String prefixChars(int prefixLen) {
-        return this.toHexString().substring(0, prefixLen);
-    }
+    String prefixChars(int prefixLen);
 
 
     /**
      * Returns the first 6 hexadecimal digits of the {@link SecureHash} value.
      */
     @NotNull
-    public String prefixChars() {
-        return prefixChars(6);
-    }
-
-    /**
-     * Compares the two given instances of the {@link SecureHash} based on the content.
-     */
-    public boolean equals(@Nullable Object other) {
-        if (this == other) return true;
-        return other instanceof SecureHash && this.algorithm.equals(((SecureHash) other).algorithm) && super.equals(other);
-    }
-
-    /**
-     * Returns a hash code value for the object.
-     */
-    public int hashCode() {
-        return ByteBuffer.wrap(this.getBytes()).getInt();
-    }
+    String prefixChars();
```

Comment:
@driessamyn should we be removing this overload? Implementation is just `prefixChars(6)`

Reply:
All `prefixChars` APIs removed after internal communication.